connect/runtime/src/main/java/org/apache/kafka/connect/runtime/errors/RetryWithToleranceOperator.java

```diff
@@ -175,9 +179,8 @@ protected <V> V execAndRetry(Operation<V> operation) throws Exception {
                 log.trace("Caught a retriable exception while executing {} operation with {}", context.stage(), context.executingClass());
                 errorHandlingMetrics.recordFailure();
                 if (checkRetry(startTime)) {
-                    backoff(attempt, deadline);
-                    if (Thread.currentThread().isInterrupted()) {
-                        log.trace("Thread was interrupted. Marking operation as failed.");
+                    if (!backoff(attempt, deadline) || Thread.currentThread().isInterrupted()) {
```

Comment:
We don't need the check to see if the thread is interrupted anymore; this was only included because of the semantics for `SystemTime::sleep`, which in turn used [Utils:sleep](https://github.com/apache/kafka/blob/5eff8592cc0cb57475c92bce6e80264d71a9bfbe/clients/src/main/java/org/apache/kafka/common/utils/Utils.java#L372-L383), which handled interruptions by resetting the thread interrupt flag and then returning normally.